pkg/kubelet/nodelease/controller.go

```diff
@@ -60,11 +60,17 @@ type controller struct {
 }
 
 // NewController constructs and returns a controller
-func NewController(clock clock.Clock, client clientset.Interface, holderIdentity string, leaseDurationSeconds int32, onRepeatedHeartbeatFailure func()) Controller {
+func NewController(clock clock.Clock, client clientset.Interface, holderIdentity string, leaseDurationSeconds int32, nodeStatusUpdateFrequency time.Duration, onRepeatedHeartbeatFailure func()) Controller {
 	var leaseClient coordclientset.LeaseInterface
 	if client != nil {
 		leaseClient = client.CoordinationV1().Leases(corev1.NamespaceNodeLease)
 	}
+	var renewInterval time.Duration
```

Comment:
is there ever a valid case for renewInterval being >= leaseDurationSeconds? 

protecting against a mismatch between leaseDurationSeconds and nodeStatusUpdateFrequency by forcing renewInterval to be at least X seconds before or < X% of leaseDurationSeconds would make sense to me

Reply:
personally, not found that case. 

IIUC, the `leaseDurationSeconds` seems to be a redundancy filed and not been used by node healthy-related work at least for now. 
As may be used in the future, I agree we should have some restriction to make renewInterval being smaller than `leaseDurationSeconds`

Reply:
> forcing renewInterval to be at least X seconds before or < X% of leaseDurationSeconds

I found it difficult to generate a smart enough `renewInterval` value based on the `leaseDurationSeconds`, because different user may have different `leaseDurationSeconds`. 

As  `leaseDurationSeconds` have not been used by node healthy-related work for now, we may ignore this restriction. only forcing this value based on `nodeStatusUpdateFrequency` is enough.   

so we can cherry-pick this PR to v1.14 and 1.15 safely.

After this fix, we should add a new field to KubeletConfig at head, let user customize the `renewInterval`  value based on their own circumstance. we don't need to worry about this value anymore other than write some documents.

Reply:
@wojtek-t  what do you think about my proposal?

Reply:
> is there ever a valid case for renewInterval being >= leaseDurationSeconds?

No - I don't see any usecase for that.
But yeah - as @gaorong pointed out, leaseDuriationSeconds is currently only propagated as part of Lease object and not really used in any meaningful way.

I'm personally not 100% convinced that connecting "renewInterval" with "nodeStatusUpdateFrequency" is exactly what we want. In the ideal world, this should be connected with "node-monitor-grace-period", but that's not even parameter to kubelet, so we don't have any way to reasonably validate it.

So as a workaround, I'm temporarily fine with the logic you proposed in the PR, if we add a very explicit comment of why we're doing that.
So let's change it something like:

```
renewInterval := defaultRenewInterval
// Users are able to decrease the timeout after which nodes are being
// marked as Ready: Unknown by NodeLifecycleController to values
// smaller than defaultRenewInterval. Until the knob to configure
// lease renew interval is exposed to user, we temporarily decrease
// renewInterval based on the NodeStatusUpdateFrequency.
if renewInterval > nodeStatusUpdateFrequency {
  renewInterval = nodeStatusUpdateFrequency
}
```

In the meantime if you could also work on exposing it via kubelet config, it would be great.

Reply:
changed as comments.

I'm going to working on exposing it via kubelet config.